src/sql/workbench/contrib/notebook/browser/notebookLinkHandler.ts

```diff
@@ -118,6 +118,34 @@ export class NotebookLinkHandler {
 		}
 	}
 
+	/**
+	* Function to get encoded url,
```

Comment:
`Gets the link URI-encoded link URL (https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/encodeURI)`

(I don't think it's necessary to call out that we check it's already encoded - that is more of an implementation detail and should be assumed since not doing so would be a bug generally)